iceoryx_posh/test/moduletests/test_roudi_iceoyx_roudi_memory_manager.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "test.hpp"
+
+#include "iceoryx_posh/roudi/memory/iceoryx_roudi_memory_manager.hpp"
+
+using namespace ::testing;
+
+using iox::roudi::IceOryxRouDiMemoryManager;
+
+namespace iox
+{
+namespace test
+{
+
+/// @brief This test file verifies that the BaseClass IceoryxRouDiMemoryManager is tested
+class IceoryxRoudiMemoryManager_test : public Test
+{
+  public:
+    std::unique_ptr<IceOryxRouDiMemoryManager> m_roudiMemoryManagerTest;
+
+    void SetUp() override
+    {
+      auto config = iox::RouDiConfig_t().setDefaults();
+      m_roudiMemoryManagerTest = std::unique_ptr<IceOryxRouDiMemoryManager>(new IceOryxRouDiMemoryManager(config));
+    }
+
+    void TearDown() override
+    {
+    }
+};
+
+/// @brief test to check function introspectionMemoryManager
+TEST_F(IceoryxRoudiMemoryManager_test, IntrospectionMemoryManagerNotInitialized)
```

Comment:
I would rename all those test cases to xxNulloptWhenNot present.

What about the test cases for when they are present? This means when the `RouDiMemoryManager` has created them.
I think we need the following additional test cases:
- In ctor `m_memoryManager.addMemoryProvider` fails and `std::terminate` is called
- `createAndAnnounceMemory`
   * `m_memoryManager` called `createAndAnnounce` with success and once failed
   * `m_portPoolBlock.portPool()` failed.
   * calling it multiple times
   * calling it in combination with `destroyMemory`
- `destroyMemory` calling it once and multiple times.

The reason why I am mention those tests are that we have to test everything with MC/DC in mind when we go for safety. Take a look at the coffee and cup example here: https://www.rapitasystems.com/mcdc-coverage

Reply:
I understood your point. But unfortunately, in the cases you are proposing some cannot be tested in unit test. But, I will add the one which can be.

Reply:
@JimmyBch would be great if you add those that can be tested. There are some things which we currently can't test and need to think how to adjust our code to make it testable as well as not uglify it.
In this special case with the ctor, I think it's sufficient to test the happy path, since if that works in the test, it will never fail.

Reply:
I added new tests cases. In my opinion, I cannot add more. Indeed, from my understanding error tests cannot be hit by unit test.

Reply:
@JimmyBch you can check if `mgmtMemoryProvider()` doesn't return a `nullptr`
Also, after calling `createAndAnnounceMemory` all the methods returning a `cxx::optional` must not return a `nullopt` anymore and accessing the objects, e.g. by calling `getPublisherPortDataList` on the `PortPool` or `getChunk` on the introspection `MemoryManager` should not cause segfaults.
And at last, after calling `destroyMemory` all of those accesses should cause segfaults which can be tested with a `DEATH_TEST`.